workflows/kfdrc-ont-longreads-workflow.cwl

```diff
@@ -358,10 +327,160 @@ steps:
+    in:
+      input_bam: minimap2/out_alignments
+      indexed_reference_fasta: indexed_reference_fasta
+      output_basename:
+        source: output_basename
+        valueFrom: $(self).nanocaller
+      mode:
+        valueFrom: "snps"
+      phase:
+        valueFrom: |
+          $(1 == 1)
+      wgs_contigs: nanocaller_wgs_contigs
+      regions: nanocaller_regions
+      include_bed: nanocaller_include_bed
+      exclude_bed: nanocaller_exclude_bed
+      exclude_bed_preset: nanocaller_exclude_bed_preset
+      preset: nanocaller_preset
+      snp_model: nanocaller_snp_model
+      neighbor_threshold: nanocaller_neighbor_threshold
+      mincov:
+        source: [nanocaller_mincov, samtools_coverage_1/meandepth]
+        valueFrom: |
+          $(self[0] != null ? self[0] : Math.ceil(self[1]/4))
+      maxcov: nanocaller_maxcov
+      haploid_X:
+        source: [sex, samtools_coverage_1/meandepth, samtools_coverage_x/meandepth]
+        valueFrom: |
+          $(self[0] == "male" ? true : self[2]/self[1] < 0.75)
+      cpu:
+        valueFrom: $(36)
+      ram:
+        valueFrom: $(36)
+    out: [snps_unphased_vcf, snps_phased_vcf, indels_vcf, final_vcf, phased_bams]
+  nanocaller_indels_diploid:
+    run: ../tools/nanocaller.cwl
+    scatter: [input_bam]
+    hints:
+    - class: sbg:AWSInstanceType
+      value: c5.12xlarge
+    in:
+      input_bam: nanocaller_snps/phased_bams
+      indexed_reference_fasta: indexed_reference_fasta
+      output_basename:
+        source: output_basename
+        valueFrom: $(self).nanocaller
+      mode:
+        valueFrom: "indels"
+      regions:
+        valueFrom: |
+          $([inputs.input_bam.basename.split('.')[0]])
+      include_bed: nanocaller_include_bed
+      exclude_bed: nanocaller_exclude_bed
+      exclude_bed_preset: nanocaller_exclude_bed_preset
+      preset: nanocaller_preset
+      indel_model: nanocaller_indel_model
+      ins_threshold: nanocaller_ins_threshold
+      del_threshold: nanocaller_del_threshold
+      win_size: nanocaller_win_size
+      small_win_size: nanocaller_small_win_size
+      mincov:
+        source: [nanocaller_mincov, samtools_coverage_1/meandepth]
+        valueFrom: |
+          $(self[0] != null ? self[0] / 2 : Math.ceil(self[1]/8))
+      maxcov: nanocaller_maxcov
+      cpu:
+        valueFrom: $(8)
+      ram:
+        valueFrom: $(8)
+    out: [snps_unphased_vcf, snps_phased_vcf, indels_vcf, final_vcf, phased_bams]
+  nanocaller_indels_haploid:
+    run: ../tools/nanocaller.cwl
+    hints:
+    - class: sbg:AWSInstanceType
+      value: c5.12xlarge
+    in:
+      input_bam: minimap2/out_alignments
+      indexed_reference_fasta: indexed_reference_fasta
+      output_basename:
+        source: output_basename
+        valueFrom: $(self).nanocaller
+      mode:
+        valueFrom: "indels"
+      regions:
+        source: [sex, samtools_coverage_1/meandepth, samtools_coverage_x/meandepth]
+        valueFrom: |
```

Comment:
interesting...

Reply:
Yeah what are your thoughts on this? Since Nanocaller is the only place that sex is used I wonder if I should change that to just asking if its better to just ask if they know if the X is diploid. Like if the user has an XXY male, I feel like they'll be like "I didn't know this was how you were going to use that input! I knew X was diploid!"